tests/checkers/unittest_variables.py

```diff
@@ -22,7 +22,7 @@ class TestVariablesChecker(CheckerTestCase):
 
     def test_all_elements_without_parent(self) -> None:
         node = astroid.extract_node("__all__ = []")
-        node.value.elts.append(astroid.Const("test"))
+        node.value.elts.append(astroid.Const("test", parent=None))
         root = node.root()
         with self.assertNoMessages():
             self.checker.visit_module(root)
```

Comment:
I don't think it's a good test. Why should adding a constant not result in messages? That's just asserting implementation quirks. We should test the opposite, we should test that synthetic constants result in the same messages.
IMO